src/executionTimingsReporter.ts

```diff
@@ -0,0 +1,139 @@
+import { HighResTimer, calculateElapsedTime } from "./utils";
+import { EventEmitter } from "events";
+
+export type TimingsReport = {[stepName: string]: [number] | number};
+
+export const stepStartedEventName = 'stepStarted';
+export const milestoneReachedEventName = 'milestoneReached';
+
+export interface StepStartedEventArguments {
+    stepName: string;
+}
+
+export interface MilestoneReachedEventArguments {
+    milestoneName: string;
+}
+
+export interface ObservableEvents<T> { // Normally T will be an EventEmitter
+    Events: T;
+}
+
+export interface StepStartedEventsEmitter {
+    on(event: 'stepStarted', listener: (args: StepStartedEventArguments) => void): this;
+    on(event: 'milestoneReached', listener: (args: MilestoneReachedEventArguments) => void): this;
+}
+
+export interface NavigatedToUserRequestedUrlEventsEmitter {
+    on(event: 'navigatedToUserRequestedUrl', listener: () => void): this;
+    once(event: 'navigatedToUserRequestedUrl', listener: () => void): this;
+}
+
+export class StepProgressEventsEmitter extends EventEmitter {
+    constructor(private readonly _nestedEmitters: [StepStartedEventsEmitter] = [] as [StepStartedEventsEmitter]) {
+        super();
+    }
+
+    public emitStepStarted(stepName: string): void {
+        this.emit(stepStartedEventName, { stepName: stepName } as StepStartedEventArguments);
+    }
+
+    public emitMilestoneReached(milestoneName: string): void {
```

Comment:
Is this used anywhere?

Reply:
This will be used in configurationDone in vscode-chrome-debug